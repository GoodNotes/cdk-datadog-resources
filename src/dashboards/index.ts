export * from './datadog-dashboard-schema.generated';
export * from './datadog-api-dashboard.generated';
export * from './datadog-dashboard';
