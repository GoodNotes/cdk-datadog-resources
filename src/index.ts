export * from './monitors';
export * from './dashboards';
export * from './slos';
