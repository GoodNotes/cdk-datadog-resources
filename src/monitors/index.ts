export * from './datadog-monitor-schema.generated';
export * from './datadog-monitor';
export * from './properties';
