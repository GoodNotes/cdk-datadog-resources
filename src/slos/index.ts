export * from './datadog-slo-schema.generated';
export * from './datadog-slo';
export * from './properties';
