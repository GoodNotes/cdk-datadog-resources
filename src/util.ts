import * as decamelize from "decamelize";

export function snakeCaseKeys<T = unknown>(obj: T, sep = "_"): T {
  if (typeof obj !== "object" || obj == null) {
    return obj;
  }

  if (Array.isArray(obj)) {
    return obj.map((o) => snakeCaseKeys(o, sep)) as any;
  }

  const result: Record<string, unknown> = {};
  for (let [k, v] of Object.entries(obj)) {
    if (typeof v === "object" && v != null) {
      v = snakeCaseKeys(v);
    }
    result[decamelize(k, { separator: sep })] = v;
  }
  return result as any;
}
