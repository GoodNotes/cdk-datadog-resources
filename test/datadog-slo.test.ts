import { Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { DatadogSLO, SLOType, TimeframeType } from '../src';

test('Snapshot test', () => {
  const stack = new Stack();

  new DatadogSLO(stack, 'TestSLO', {
    name: 'Example-Service-Level-Objective',
    description: 'Example SLO',
    groups: ['env:test', 'role:mysql'],
    tags: ['env:prod', 'app:core'],
    type: SLOType.METRIC,
    query: {
      denominator: 'sum:httpservice.hits{!code:3xx}.as_count()',
      numerator: 'sum:httpservice.hits{code:2xx}.as_count()',
    },
    thresholds: [
      {
        target: 97.0,
        targetDisplay: '97.0',
        timeframe: TimeframeType.SEVENDAYS,
        warning: 98,
        warningDisplay: '98.0',
      },
    ],
  });
  expect(Template.fromStack(stack)).toMatchSnapshot();
});
