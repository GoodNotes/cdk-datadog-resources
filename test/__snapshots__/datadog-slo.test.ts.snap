// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "TestSLO": Object {
      "Properties": Object {
        "Description": "Example SLO",
        "Groups": Array [
          "env:test",
          "role:mysql",
        ],
        "Name": "Example-Service-Level-Objective",
        "Query": Object {
          "Denominator": "sum:httpservice.hits{!code:3xx}.as_count()",
          "Numerator": "sum:httpservice.hits{code:2xx}.as_count()",
        },
        "Tags": Array [
          "env:prod",
          "app:core",
        ],
        "Thresholds": Array [
          Object {
            "Target": 97,
            "TargetDisplay": "97.0",
            "Timeframe": "7d",
            "Warning": 98,
            "WarningDisplay": "98.0",
          },
        ],
        "Type": "metric",
      },
      "Type": "Datadog::SLOs::SLO",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
